pydantic/main.py

```diff
@@ -810,6 +564,383 @@ def model_parametrized_name(cls, params: tuple[type[Any], ...]) -> str:
+            exclude=exclude,
+            by_alias=by_alias,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+        )
+
+    @classmethod
+    def parse_obj(cls: type[Model], obj: Any) -> Model:
+        warnings.warn('The `parse_obj` method is deprecated; use `model_validate` instead.', DeprecationWarning)
+        return cls.model_validate(obj)
+
+    @classmethod
+    def parse_raw(
+        cls: type[Model],
+        b: str | bytes,
+        *,
+        content_type: str = None,
+        encoding: str = 'utf8',
+        proto: _deprecated_parse.Protocol = None,
+        allow_pickle: bool = False,
+    ) -> Model:
+        warnings.warn(
+            'The `parse_raw` method is deprecated; load the data and use `model_validate` instead.', DeprecationWarning
+        )
+        try:
+            obj = _deprecated_parse.load_str_bytes(
+                b,
+                proto=proto,
+                content_type=content_type,
+                encoding=encoding,
+                allow_pickle=allow_pickle,
+                json_loads=cls.model_config['json_loads'],
+            )
+        except (ValueError, TypeError, UnicodeDecodeError) as e:
+            # TODO: this used to raise ValidationError([ErrorWrapper(e, loc=ROOT_KEY)], cls)
+            #   Do we want to make it possible to instantiate from python, just for deprecated functionality?
+            raise e
+        return cls.parse_obj(obj)
+
+    @classmethod
+    def parse_file(
+        cls: type[Model],
+        path: str | Path,
+        *,
+        content_type: str = None,
+        encoding: str = 'utf8',
+        proto: _deprecated_parse.Protocol = None,
+        allow_pickle: bool = False,
+    ) -> Model:
+        warnings.warn(
+            'The `parse_file` method is deprecated; load the data and use `model_validate` instead.', DeprecationWarning
+        )
+        obj = _deprecated_parse.load_file(
+            path,
+            proto=proto,
+            content_type=content_type,
+            encoding=encoding,
+            allow_pickle=allow_pickle,
+            json_loads=cls.model_config['json_loads'],
+        )
+        return cls.parse_obj(obj)
+
+    @classmethod
+    def from_orm(cls: type[Model], obj: Any) -> Model:
+        warnings.warn(
+            'The `from_orm` method is deprecated; set model_config["from_attributes"]=True '
+            'and use `model_validate` instead.',
+            DeprecationWarning,
+        )
+        if not cls.model_config['from_attributes']:
+            raise PydanticUserError('You must set the config attribute `from_attributes=True` to use from_orm')
+        return cls.model_validate(obj)
+
+    @classmethod
+    def construct(cls: type[Model], _fields_set: set[str] | None = None, **values: Any) -> Model:
+        warnings.warn('The `construct` method is deprecated; use `model_construct` instead.', DeprecationWarning)
```

Comment:
I have not copied over the v1 docstrings. Is that okay, or do we want to do that? I'm hoping the deprecation warning pointing the user at the other method is enough?

Reply:
Agreed.